File: nelio-alves-algoritmo-e-logica-de-programacao/cpp/exercicios/soma_matrizes.cpp
#include <iostream>

using namespace std;

int main() {
    int M, N;

    cout << "Quantas linhas vai ter cada matriz? ";
    cin >> M;
    cout << "Quantas colunas vai ter cada matriz? ";
    cin >> N;

    int A[M][N], B[M][N], C[M][N];

    cout << "Digite os valores da matriz A:" << endl;
    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> A[i][j];
        }
    }

    cout << "Digite os valores da matriz B:" << endl;
    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> B[i][j];
        }
    }

    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            C[i][j] = A[i][j] + B[i][j];
        }
    }

    cout << endl << "MATRIZ A:" << endl;
    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            cout << A[i][j] << " ";
        }
        cout << endl;
    }

    cout << endl << "MATRIZ B:" << endl;
    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            cout << B[i][j] << " ";
        }
        cout << endl;
    }

    cout << endl << "MATRIZ SOMA:" << endl;
    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            cout << C[i][j] << " ";
        }
        cout << endl;
    }

    return 0;
}

File: nelio-alves-algoritmo-e-logica-de-programacao/python/exercicios/soma_matrizes.py
M: int; N: int

M = int(input("Quantas linhas vai ter cada matriz? "))
N = int(input("Quantas colunas vai ter cada matriz? "))

A: [[int]] = [[0 for x in range(N)] for x in range(M)]
B: [[int]] = [[0 for x in range(N)] for x in range(M)]
C: [[int]] = [[0 for x in range(N)] for x in range(M)]

print("Digite os valores da matriz A:")
for i in range(M):
	for j in range(N):
		A[i][j] = int(input(f"Elemento [{i},{j}]: "))

print("Digite os valores da matriz B:")
for i in range(M):
	for j in range(N):
		B[i][j] = int(input(f"Elemento [{i},{j}]: "))

for i in range(M):
	for j in range(N):
		C[i][j] = A[i][j] + B[i][j]

print("\nMATRIZ A:")
for i in range(M):
	for j in range(N):
		print(f"{A[i][j]} ", end="")
	print()

print("\nMATRIZ B:")
for i in range(M):
	for j in range(N):
		print(f"{B[i][j]} ", end="")
	print()

print("\nMATRIZ SOMA:")
for i in range(M):
	for j in range(N):
		print(f"{C[i][j]} ", end="")
	print()